dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/hibernate/HibernateProgramStageInstanceStore.java

```diff
@@ -111,15 +112,23 @@ public long getProgramStageInstanceCountLastUpdatedAfter( Date time )
     @Override
     public boolean exists( String uid )
     {
-        Integer result = jdbcTemplate.queryForObject( "select count(*) from programstageinstance where uid=? and deleted is false", Integer.class, uid );

```

Comment:
Could we add some tests here to verify that the exists methods work?

Reply:
yes, we have that. Here https://github.com/dhis2/dhis2-core/blob/d9ffd25026643592322df196b09bf5b542eddba0/dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/program/ProgramStageInstanceStoreTest.java#L254